app/controllers/repo_controller.rb

```diff
@@ -78,7 +78,16 @@ def addQuestions(client, repo, path)
       # If .JSON file is found, keep the path
       if file.name == "info.json"
         question_name = file.path.partition('/')[2].rpartition('/')[0]
-        Question.create(title: question_name, repo: repo)
+
+        require 'json'
+        path_str = path+"/info.json"
+
+        contentsJsonInfobase64 = client.contents(repo, path: path_str)
+
+        require "base64"
+        descriptive_title = JSON.parse(Base64.decode64(contentsJsonInfobase64[:content]))["title"]
+
+        Question.create(title: question_name, descriptiveTitle: descriptive_title, repo: repo)
```

Comment:
* we should use camel_case names
* will need to fix `descriptivetitle`
* Ideally there are test cases which cover this
* 
Arguably, ` JSON.parse(Base64.decode64(info_json_base64[:content]))["title"]` should be a method:

```rb
def extract_title_from_file(encoded_file)
   JSON.parse(Base64.decode64(encoded_file))["title"]
end
```

```rb
        descriptive_title = extract_title_from_file(encoded_file[:content])
```

```suggestion
        encoded_file = client.contents(repo, path: path_str)
        descriptive_title = extract_title_from_file(encoded_file[:content])
        Question.create(title: question_name, descriptivetitle: descriptive_title, repo: repo)
```